src/settings/SettingsScreen.js

```diff
@@ -47,11 +49,27 @@ export default function SettingsScreen(props: Props): Node {
           dispatch(setGlobalSettings({ browser: value ? 'embedded' : 'external' }));
         }}
       />
-      <SwitchRow
-        label="Do not mark messages read on scroll"
-        value={doNotMarkMessagesAsRead}
+      <InputRowRadioButtons
+        navigation={navigation}
+        label="Mark messages as read on scroll"
+        description="When scrolling through messages, should they automatically be marked as read?"
+        valueKey={(markMessagesReadOnScroll: number)}
+        items={[
+          { key: (MarkMessagesReadOnScroll.Always: number), title: 'Always' },
+          { key: (MarkMessagesReadOnScroll.Never: number), title: 'Never' },
+          {
+            key: (MarkMessagesReadOnScroll.ConversationViewsOnly: number),
+            title: 'Only in conversation views',
+            subtitle:
+              'Messages will be marked as read in single-topic or private-message views, but not in interleaved views, such as whole-stream views.',
+          },
+        ]}
         onValueChange={value => {
-          dispatch(setGlobalSettings({ doNotMarkMessagesAsRead: value }));
+          dispatch(
+            setGlobalSettings({
+              markMessagesReadOnScroll: MarkMessagesReadOnScroll.cast(value),
```

Comment:
Hmm, awkward to have to do this. (Also the static casts `(…: number)` above.)

It looks like the root of this is that we say `key={item.key}` in `SelectableOptionsScreen`, passing these keys as React `key` pseudoprops. Ideally the type accepted there could be like "string, or number, or any enum of string or number", but there's no way to express the "any enum" part:
https://flow.org/en/docs/enums/using-enums/#toc-generic-enums
so it isn't.

I think in light of this bit I'd lean toward skipping the enum here, and sticking with the disjoint union of strings. There isn't a lot that we gain in this instance from the enum. But this price for using an enum isn't too bad, either.